quantum/rgblight.c

```diff
@@ -677,33 +683,39 @@ void rgblight_sethsv_slave(uint16_t hue, uint8_t sat, uint8_t val) {
 
 #ifndef RGBLIGHT_CUSTOM_DRIVER
 void rgblight_set(void) {
-  LED_TYPE *start_led = led + clipping_start_pos;
+  LED_TYPE led_frame[RGBLED_NUM];
+  LED_TYPE *start_led = led_frame + clipping_start_pos;
   uint16_t num_leds = clipping_num_leds;
   if (rgblight_config.enable) {
-    #ifdef RGBLIGHT_LED_MAP
-      LED_TYPE led0[RGBLED_NUM];
-      for(uint8_t i = 0; i < RGBLED_NUM; i++) {
-          led0[i] = led[pgm_read_byte(&led_map[i])];
-      }
-      start_led = led0 + clipping_start_pos;
-    #endif
-    #ifdef RGBW
-      ws2812_setleds_rgbw(start_led, num_leds);
-    #else
-      ws2812_setleds(start_led, num_leds);
-    #endif
+    for (uint8_t i = 0; i < RGBLED_NUM; i++) {
+      #ifdef RGBLIGHT_LED_MAP
+      uint8_t src_led_idx = pgm_read_byte(&led_map[i]);
+      #else
+      uint8_t src_led_idx = i;
+      #endif
+      led_frame[i].r = pgm_read_byte(&CIE1931_CURVE[led[src_led_idx].r]);
```

Comment:
Need to add #ifdef USE_CIE1931_CURVE check around the use of the curve as it can be defined out.

Reply:
No it can't. `common_features.mk` defines `CIE1931_CURVE` when `RGBLIGHT_ENABLE=yes`, and `rgblight.c.` breaks when it is not there. And I don't think this should be supported; RGB LEDs are quite sad without gamma correction. Try disabling gamma correction and see how it looks.

Reply:
oh I disabled it once before easily enough, though I was using rgb matrix at the time. And while common_features.mk might define it, it can always be undefined easily enough from user's mk or code. Though I do agree it is silly to undefine it. My main gripe is that where the table is defined is wrapped with that check, but where we use it it isn't. So we should at least fix our usage or fix the table.

Reply:
Fair enough. And here, I can also use one of my own arguments against me: in a hypothetical future where there exists a keyboard using gamma-corrected LEDs, there's no reason for QMK to store and use gamma table!